src/utils/components/SelectTypeahead/SelectTypeahead.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { FC, ReactNode, Ref, useRef, useState } from 'react';
+
+import {
+  MenuToggleElement,
+  Select,
+  SelectList,
+  SelectOption,
+  SelectOptionProps,
+} from '@patternfly/react-core';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { isEmpty } from '@utils/utils';
+
+import Toggle from './Toggle';
+import { filterOptions } from './utils';
+
+type SelectTypeaheadProps = {
+  canCreate?: boolean;
+  id: string;
+  newOptionComponent?: (inputValue: string) => ReactNode;
+  options: SelectOptionProps[];
+  placeholder: string;
+  selected: string;
+  setSelected: (newSelection: null | string) => void;
+};
+
+const SelectTypeahead: FC<SelectTypeaheadProps> = ({
+  canCreate = true,
+  id,
+  newOptionComponent,
+  options,
+  placeholder,
+  selected,
+  setSelected,
+}) => {
+  const { t } = useNetworkingTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  const [inputValue, setInputValue] = useState<string>(selected || '');
+  const [focusedItemIndex, setFocusedItemIndex] = useState<null | number>(null);
+  const textInputRef = useRef<HTMLInputElement>();
+
+  const selectOptions = inputValue ? filterOptions(options, inputValue) : options;
+
+  const onSelect = (value: string) => {
```

Comment:
it looks like you're missing handling the flow of clicking the new option you can create in this type of select input